x/emissions/keeper/queryserver/query_server_inferences.go

```diff
@@ -80,10 +83,76 @@ func (qs queryServer) GetLatestNetworkInference(
 		return nil, err
 	}
 
-	return &types.QueryLatestNetworkInferencesAtBlockResponse{
+	return &types.QueryLatestNetworkInferencesResponse{
+		NetworkInferences:         networkInferences,
+		InfererWeights:            synth.ConvertWeightsToArrays(infererWeights),
+		ForecasterWeights:         synth.ConvertWeightsToArrays(forecasterWeights),
+		ForecastImpliedInferences: synth.ConvertForecastImpliedInferencesToArrays(forecastImpliedInferenceByWorker),
+		InferenceBlockHeight:      inferenceBlockHeight,
+		LossBlockHeight:           lossBlockHeight,
+	}, nil
+}
+
+func (qs queryServer) GetLatestNetworkInferenceWithLosses(
+	ctx context.Context,
+	req *types.QueryLatestNetworkInferencesRequest,
+) (
+	*types.QueryLatestNetworkInferencesResponse,
+	error,
+) {
+	topic, err := qs.k.GetTopic(ctx, req.TopicId)
+	if err != nil {
+		return nil, err
+	}
+
+	inferences, inferenceBlockHeight, err := qs.k.GetLatestTopicInferences(ctx, req.TopicId)
+	if err != nil {
+		return nil, err
+	}
+
+	previousLossBlockHeight := inferenceBlockHeight - topic.EpochLength
+
+	for true {
+		if previousLossBlockHeight < 0 {
+			return nil, status.Errorf(codes.NotFound, "losses not yet available for topic %v", req.TopicId)
+		}
```

Comment:
We can have this case and also a case where losses are available but there isn't a subsequent inference. Since this is a 'strict' query, I think we should also cover this case.

Reply:
Hmm this should be handled with the line `if len(inferences.Inferences) > 0 {` 

right?

Reply:
ah with some more tests I see the edge case. Working on a fix

Reply:
fixed!